ouroboros-consensus-test/bench-mempool/Bench/Consensus/Mempool/Params.hs

```diff
@@ -0,0 +1,181 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE TupleSections    #-}
+
+{-# OPTIONS_GHC -Wno-unrecognised-pragmas #-}
+{-# HLINT ignore "Use camelCase" #-}
+
+module Bench.Consensus.Mempool.Params (
+    -- * Types
+    InitialMempoolAndModelParams (..)
+  , mkParams
+    -- * Defaults
+  , defaultInMemoryBSS
+  , defaultLMDB_BSS
+  , defaultLedgerDbCfg
+  , defaultParams
+    -- * Construction of configurations
+  , sampleLedgerConfig
+  , sampleLedgerDbCfg
+    -- * Construction of initial state
+  , ledgerStateFromTokens
+  , mkSimpleGenTx
+  , mkSimpleGenesisGenTx
+  , mkSimpleGenesisTx
+  , mkSimpleTx
+  , testBlocksFromTxs
+  ) where
+
+import           Bench.Consensus.Mempool.TestBlock (GenTx (TestBlockGenTx),
+                     PayloadDependentState (..), TestBlock, Token (..), Tx)
+import qualified Bench.Consensus.Mempool.TestBlock as Mempool.TestBlock
+import qualified Cardano.Slotting.Time as Time
+import           Control.Monad.IO.Class
+import qualified Data.Map.Strict as Map
+import           Data.Maybe (maybeToList)
+import qualified Ouroboros.Consensus.Block as Block
+import           Ouroboros.Consensus.Config.SecurityParam
+                     (SecurityParam (SecurityParam))
+import qualified Ouroboros.Consensus.HardFork.History as HardFork
+import           Ouroboros.Consensus.Ledger.Basics (LedgerConfig, ValuesMK (..))
+import           Ouroboros.Consensus.Storage.LedgerDB (BackingStoreSelector,
+                     LedgerDbCfg)
+import           Ouroboros.Consensus.Storage.LedgerDB.BackingStore.LMDB
+                     (LMDBLimits (..))
+import           Ouroboros.Consensus.Storage.LedgerDB.Init
+                     (BackingStoreSelector (..))
+import           Ouroboros.Consensus.Storage.LedgerDB.LedgerDB
+                     (LedgerDbCfg (..))
+import qualified Test.Util.TestBlock as TestBlock
+import           Test.Util.TestBlock (LedgerState (..))
+
+{-------------------------------------------------------------------------------
+  Types
+-------------------------------------------------------------------------------}
+
+-- | Initial parameters for the mempool benchmarks.
+--
+-- === Parameters for the ledger interface
+--
+-- One goal of the parameters is to provide enough information to set up an
+-- interface to the ledger (database). How we populate the ledger database with
+-- values and differences could affect the performance of mempool operations. To
+-- be precise, each time we need a partial ledger state to apply transactions
+-- to, we /rewind-read-forward/.
+--
+-- * Rewind: Rewind keys by determining which slot the tip of the backing store
+--   points to.
+-- * Read: Read values from the backing store for the rewound keys.
+-- * Forward: Forward the read values through the changelog.
+--
+-- How expensive these steps are depends on how much initial state we set up in
+-- the backing store and changelog.
+--
```

Comment:
I was intentionally a bit vague here because I'm not sure if we can estimate the cost on these parameters only, but yes a small `immpBlocks` does mean a cheaper forward in practice

Reply:
So perhaps we could simply mention that :)